ADVANCED_EXAMPLES.md

```diff
@@ -0,0 +1,409 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [`object.getStamp()`. 2 ways.](#objectgetstamp-2-ways)
+  - [First way](#first-way)
+  - [Second way](#second-way)
+- [Self cloneable objects. 3 ways.](#self-cloneable-objects-3-ways)
+  - [A composable stamp which adds `.clone()` to objects](#a-composable-stamp-which-adds-clone-to-objects)
+  - [Another way of self cloning](#another-way-of-self-cloning)
+  - [Memory efficient cloning](#memory-efficient-cloning)
+- [Delayed object instantiation using Promises](#delayed-object-instantiation-using-promises)
+- [Dependency injection tips](#dependency-injection-tips)
+- [Validate before a function call](#validate-before-a-function-call)
+- [EventEmitter without inheritance (`convertConstructor`)](#eventemitter-without-inheritance-convertconstructor)
+- [Hacking stamps](#hacking-stamps)
+  - ["Default" properties](#default-properties)
+  - ["Default" properties as composable behavior](#default-properties-as-composable-behavior)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+## `object.getStamp()`. 2 ways.
+
+> Run the examples below for yourself:
+```sh
+$ git clone https://github.com/stampit-org/stampit.git
+$ node stampit/advanced-examples/self-aware.js
+```
+
+You can add `.getStamp()` function to each object with ease. 
+
+First, let's assume you have a following stamp:
+```js
+var User = stampit();
+```
+
+### First way
+Just compose the following stamp to any other stamp.
+```js
+var SelfAware1 = stampit.init(function (ctx) { // the context, has `stamp` property
```

Comment:
The `node stampit/advanced-examples/self-aware.js` won't work then. :)
I wanted to make sure people can run examples without installing anything.